server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -575,6 +575,13 @@ public final class IndexSettings {
         Property.InternalIndex
     );
 
+    public static final Setting<Integer> SEARCHABLE_SNAPSHOT_MINIMUM_VERSION = Setting.intSetting(
```

Comment:
Why is this configurable? Isn't there an inherent limit as to what the software can actually support?

Reply:
Yes, there is an inherent limit, and modeling this as a setting makes it seem configurable when it really isn't. The purpose was to embed the minimum version value into index settings (as per [this comment](https://github.com/opensearch-project/OpenSearch/pull/5429#discussion_r1037656013)) so that this logic was limited to searchable snapshots ([code](https://github.com/kartg/OpenSearch/blob/94a9ad0e9787b141eb8047661c56346f61ee56ba/server/src/main/java/org/opensearch/snapshots/RestoreService.java#L1256-L1270)). As shown in that code snippet, the actual value of the setting is hard-coded.

Any suggestions on how I could make this less ugly?

Reply:
The _version_ is not a property of the index itself; it's a property of the software, which can potentially change without the data the makes up the index changing. So I don't think it makes sense as an index setting. Can you change this to pass the hard-coded version value when you construct the `ReadOnlyEngine` instance? That is already inside a "if remote snapshot" block so that seems like it's just moving the hard-coded behavior and gets rid of the index setting.